cmd/gen-key/ecdsa.go

```diff
@@ -0,0 +1,212 @@
+package main
+
+import (
+	"crypto"
+	"crypto/ecdsa"
+	"crypto/elliptic"
+	"encoding/asn1"
+	"errors"
+	"fmt"
+	"log"
+	"math/big"
+
+	"github.com/miekg/pkcs11"
+)
+
+var stringToCurve = map[string]*elliptic.CurveParams{
+	elliptic.P224().Params().Name: elliptic.P224().Params(),
+	elliptic.P256().Params().Name: elliptic.P256().Params(),
+	elliptic.P384().Params().Name: elliptic.P384().Params(),
+	elliptic.P521().Params().Name: elliptic.P521().Params(),
+}
+
+var curveToOIDDER = map[*elliptic.CurveParams][]byte{
+	elliptic.P224().Params(): []byte{6, 5, 43, 129, 4, 0, 33},
+	elliptic.P256().Params(): []byte{6, 8, 42, 134, 72, 206, 61, 3, 1, 7},
+	elliptic.P384().Params(): []byte{6, 5, 43, 129, 4, 0, 34},
+	elliptic.P521().Params(): []byte{6, 5, 43, 129, 4, 0, 35},
+}
+
+var oidDERToCurve = map[string]*elliptic.CurveParams{
+	"06052B81040021":       elliptic.P224().Params(),
```

Comment:
Can you add a comment with a pointer to the authoritative source for these OIDs?